### ISSUE TITLE
Accents incorrectly saved (or displayed) in metadata of geopackage

### ISSUE THREAD

**Issue by jfbourdon** [qgis/QGIS#58476]
### What is the bug or the crash?

Trying to save metadata containing accents (like "é") results in incorrect binary encoding and thus incorrect display of the characters afterwards. Seems to be similar to #47435.

### Steps to reproduce the issue

```python
lyr = QgsVectorLayer(f"point?crs=EPSG:4326", "lyr", "memory")
options = QgsVectorFileWriter.SaveVectorOptions()


# The resulting BLOB in the "description" field of the "gpkg_contents" table contains the hexadecimal "e9 61"
# Displayed as "�a" in QGIS
outpath = "C:/Temp/encoding_1.gpkg"
options.layerOptions = ["DESCRIPTION=éa"]
QgsVectorFileWriter.writeAsVectorFormatV3(lyr, outpath, QgsCoordinateTransformContext(), options)

# The resulting BLOB in the "description" field of the "gpkg_contents" table contains the hexadecimal "61 ef bf bd"
# Displayed as "a�" in QGIS
outpath = "C:/Temp/encoding_2.gpkg"
options.layerOptions = ["DESCRIPTION=aé"]
QgsVectorFileWriter.writeAsVectorFormatV3(lyr, outpath, QgsCoordinateTransformContext(), options)
```

Note that `options.fileEncoding = "UTF-8"` doesn't change the outcome. However, if I manually replace the binary content in "gpkg_contents" to "c3 a9 61" and "61 c3 a9", QGIS then correctly displays "éa" and "aé" respectively in all tested QGIS versions.

### Versions

QGIS 3.38.1 (but also tested on 3.22.8, 3.24.2, 3.26.0 and 3.28.11)
Windows 10 Enterprise 22H2

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by agiudiceandrea** [comment]
@jfbourdon, please try if the issue still occurs also using QGIS 3.38.2.

**Comment by jfbourdon** [comment]
Yep, I confirm that the issue has been fixed in QGIS 3.82.2. Thanks.

**Comment by agiudiceandrea** [comment]
The fix will be also available in the upcoming QGIS 3.34.11.
